.claude/skills/cfn-node-heap-sizer/task-mode-heap-limiter.sh

```
@@ -1,326 +1,326 @@
+    # Export updated NODE_OPTIONS
+    export NODE_OPTIONS="$node_opts"
+
+    echo "✅ NODE_OPTIONS configured: $NODE_OPTIONS" >&2
+    echo "   Mode: $mode" >&2
+    echo "   Tool: $tool_type" >&2
+    echo "   Heap: ${heap_size}MB" >&2
+    echo "   Semi-space: ${semi_space}MB" >&2
+
+    # Log configuration
+    if [[ -n "${CFN_VALIDATION_LOG_DIR:-}" ]]; then
+        local log_file="$CFN_VALIDATION_LOG_DIR/heap-config_$(date +%Y%m%d_%H%M%S).log"
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)